Several groupby tests are failing for debug build on Windows

Several tests are failing on Windows for the following profile:
- python `3.8.10` (not all the `3.8` pythons are failing, for instance, `3.8.0` doesn't fail);
- datatable debug build (release build doesn't fail);
- `/Od` compiler flag is supplied, i.e. no optimization (other optimization levels do not fail).

Here are the failing tests (https://ci.appveyor.com/project/h2oops/datatable/builds/44409622/job/9dg7sfhyqo0bu3x9?fullLog=true):

```python
_____________________________ test_group_boolean ______________________________
    def test_group_boolean():
        DT = dt.Frame(A=[True, None, False, False, True, True, False, True])
        DTR = DT[:, count(), by(f.A)]
>       assert_equals(DTR, dt.Frame(A=[None, False, True], count=[1, 3, 4],
                                    stypes={"count": dt.int64}))
DT         = <Frame#2cc0da0dcf0 8x1>
DTR        = <Frame#2cc0da0dde0 3x2>
tests\test-groups.py:247: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
frame1 = <Frame#2cc0da0dde0 3x2>, frame2 = <Frame#2cc0da0dd20 3x2>
rel_tol = 1e-07, abs_tol = 1e-07
    def assert_equals(frame1, frame2, rel_tol = 1e-7, abs_tol = None):
        """
        Helper function to assert that 2 frames are equal to each other.
        """
        frame_integrity_check(frame1)
        frame_integrity_check(frame2)
    
        # The default value of `abs_tol` is set to `rel_tol`
        if abs_tol is None:
            abs_tol = rel_tol
    
        assert frame1.shape == frame2.shape, (
            "The left frame has shape %r, while the right has shape %r"
            % (frame1.shape, frame2.shape))
    
        assert frame1.names == frame2.names, (
            "The left frame has names %r, while the right has names %r"
            % (frame1.names, frame2.names))
        assert frame1.types == frame2.types, (
            "The left frame has types %r, while the right has types %r"
            % (frame1.types, frame2.types))
        data1 = frame1.to_list()
        data2 = frame2.to_list()
        assert len(data1) == len(data2)  # shape check should ensure this
        for i in range(len(data1)):
            col1 = data1[i]
            col2 = data2[i]
            assert len(col1) == len(col2)
            for j in range(len(col1)):
                val1 = col1[j]
                val2 = col2[j]
                if val1 == val2: continue
                if isinstance(val1, float) and isinstance(val2, float):
                    if math.isclose(val1, val2, rel_tol = rel_tol, abs_tol = abs_tol): continue
                if len(col1) > 16:
                    arr1 = repr(col1[:16])[:-1] + ", ...]"
                    arr2 = repr(col2[:16])[:-1] + ", ...]"
                else:
                    arr1 = repr(col1)
                    arr2 = repr(col2)
>               raise AssertionError(
                    "The frames have different data in column %d `%s` at "
                    "index %d: LHS has %r, and RHS has %r\n"
                    "  LHS = %s\n"
                    "  RHS = %s\n"
                    % (i, frame1.names[i], j, val1, val2, arr1, arr2))
E               AssertionError: The frames have different data in column 0 `A` at index 0: LHS has False, and RHS has None
E                 LHS = [False, False, True]
E                 RHS = [None, False, True]
abs_tol    = 1e-07
arr1       = '[False, False, True]'
arr2       = '[None, False, True]'
col1       = [False, False, True]
col2       = [None, False, True]
data1      = [[False, False, True], [1, 3, 4]]
data2      = [[None, False, True], [1, 3, 4]]
frame1     = <Frame#2cc0da0dde0 3x2>
frame2     = <Frame#2cc0da0dd20 3x2>
i          = 0
j          = 0
rel_tol    = 1e-07
val1       = False
val2       = None
tests\__init__.py:140: AssertionError
_____________________________ test_group_boolean2 _____________________________
    def test_group_boolean2():
        DT = dt.Frame(A=[True, False, False] * 500 + [None, True])
        DTR = DT[:, count(), by(f.A)]
>       assert_equals(DTR, dt.Frame(A=[None, False, True], count=[1, 1000, 501],
                                    stypes={"count": dt.int64}))
DT         = <Frame#2cc0d9f65a0 1502x1>
DTR        = <Frame#2cc0d9f6600 3x2>
tests\test-groups.py:253: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
frame1 = <Frame#2cc0d9f6600 3x2>, frame2 = <Frame#2cc0d9f65d0 3x2>
rel_tol = 1e-07, abs_tol = 1e-07
    def assert_equals(frame1, frame2, rel_tol = 1e-7, abs_tol = None):
        """
        Helper function to assert that 2 frames are equal to each other.
        """
        frame_integrity_check(frame1)
        frame_integrity_check(frame2)
    
        # The default value of `abs_tol` is set to `rel_tol`
        if abs_tol is None:
            abs_tol = rel_tol
    
        assert frame1.shape == frame2.shape, (
            "The left frame has shape %r, while the right has shape %r"
            % (frame1.shape, frame2.shape))
    
        assert frame1.names == frame2.names, (
            "The left frame has names %r, while the right has names %r"
            % (frame1.names, frame2.names))
        assert frame1.types == frame2.types, (
            "The left frame has types %r, while the right has types %r"
            % (frame1.types, frame2.types))
        data1 = frame1.to_list()
        data2 = frame2.to_list()
        assert len(data1) == len(data2)  # shape check should ensure this
        for i in range(len(data1)):
            col1 = data1[i]
            col2 = data2[i]
            assert len(col1) == len(col2)
            for j in range(len(col1)):
                val1 = col1[j]
                val2 = col2[j]
                if val1 == val2: continue
                if isinstance(val1, float) and isinstance(val2, float):
                    if math.isclose(val1, val2, rel_tol = rel_tol, abs_tol = abs_tol): continue
                if len(col1) > 16:
                    arr1 = repr(col1[:16])[:-1] + ", ...]"
                    arr2 = repr(col2[:16])[:-1] + ", ...]"
                else:
                    arr1 = repr(col1)
                    arr2 = repr(col2)
>               raise AssertionError(
                    "The frames have different data in column %d `%s` at "
                    "index %d: LHS has %r, and RHS has %r\n"
                    "  LHS = %s\n"
                    "  RHS = %s\n"
                    % (i, frame1.names[i], j, val1, val2, arr1, arr2))
E               AssertionError: The frames have different data in column 0 `A` at index 0: LHS has False, and RHS has None
E                 LHS = [False, False, True]
E                 RHS = [None, False, True]
abs_tol    = 1e-07
arr1       = '[False, False, True]'
arr2       = '[None, False, True]'
col1       = [False, False, True]
col2       = [None, False, True]
data1      = [[False, False, True], [1, 1000, 501]]
data2      = [[None, False, True], [1, 1000, 501]]
frame1     = <Frame#2cc0d9f6600 3x2>
frame2     = <Frame#2cc0d9f65d0 3x2>
i          = 0
j          = 0
rel_tol    = 1e-07
val1       = False
val2       = None
tests\__init__.py:140: AssertionError
________________________ test_count_dt_groupby_integer ________________________
    def test_count_dt_groupby_integer():
        df_in = dt.Frame([9, 8, 2, 3, None, None, 3, 0, 5, 5, 8, None, 1])
>       df_reduce = df_in[:, [count(f.C0), count()], "C0"]
E       AssertionError: Assertion 'ri.max() < arg.nrows()' failed in src\core\column\view.cc, line 109
df_in      = <Frame#2cc0deef3c0 13x1>
tests\test-reduce.py:86: AssertionError
______________________ test_count_2d_dt_groupby_integer _______________________
    def test_count_2d_dt_groupby_integer():
        df_in = dt.Frame([[9, 8, 2, 3, None, None, 3, 0, 5, 5, 8, None, 1],
                          [0, 1, 0, 5, 3, 8, 1, 0, 2, 5, None, 8, 1]])
>       df_reduce = df_in[:, [count(f.C0), count(f.C1), count()], "C0"]
E       AssertionError: Assertion 'ri.max() < arg.nrows()' failed in src\core\column\view.cc, line 109
df_in      = <Frame#2cc0deb75a0 13x2>
tests\test-reduce.py:115: AssertionError
________________________ test_count_dt_groupby_string _________________________
    def test_count_dt_groupby_string():
        df_in = dt.Frame([None, "blue", "green", "indico", None, None, "orange",
                          "red", "violet", "yellow", "green", None, "blue"])
>       df_reduce = df_in[:, [count(f.C0), count()], "C0"]
E       AssertionError: Assertion 'ri.max() < arg.nrows()' failed in src\core\column\view.cc, line 109
df_in      = <Frame#2cc0dc6a450 13x1>
tests\test-reduce.py:145: AssertionError
```

Related failures with python 3.11 on Windows

```python
________________________________ test_groups1b ________________________________
    def test_groups1b():
        DT = dt.Frame([[1,   5,   3,   2,   1,    3,   1,   1,   None],
                       ["a", "b", "c", "a", None, "f", "b", "h", "d"]],
                      names=["A", "B"])
        d1 = DT[:, :, by("A")]
>       assert_equals(
            d1, dt.Frame(A=[None, 1, 1, 1, 1, 2, 3, 3, 5],
                         B=["d", "a", None, "b", "h", "a", "c", "f", "b"]))
DT         = <Frame#1afd0932f10 9x2>
d1         = <Frame#1afd0932e80 9x2>
tests\test-groups.py:44: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
frame1 = <Frame#1afd0932e80 9x2>, frame2 = <Frame#1afd0932eb0 9x2>
rel_tol = 1e-07, abs_tol = 1e-07
    def assert_equals(frame1, frame2, rel_tol = 1e-7, abs_tol = None):
        """
        Helper function to assert that 2 frames are equal to each other.
        """
        frame_integrity_check(frame1)
        frame_integrity_check(frame2)
    
        # The default value of `abs_tol` is set to `rel_tol`
        if abs_tol is None:
            abs_tol = rel_tol
    
        assert frame1.shape == frame2.shape, (
            "The left frame has shape %r, while the right has shape %r"
            % (frame1.shape, frame2.shape))
    
        assert frame1.names == frame2.names, (
            "The left frame has names %r, while the right has names %r"
            % (frame1.names, frame2.names))
        assert frame1.types == frame2.types, (
            "The left frame has types %r, while the right has types %r"
            % (frame1.types, frame2.types))
        data1 = frame1.to_list()
        data2 = frame2.to_list()
        assert len(data1) == len(data2)  # shape check should ensure this
        for i in range(len(data1)):
            col1 = data1[i]
            col2 = data2[i]
            assert len(col1) == len(col2)
            for j in range(len(col1)):
                val1 = col1[j]
                val2 = col2[j]
                if val1 == val2: continue
                if isinstance(val1, float) and isinstance(val2, float):
                    if math.isclose(val1, val2, rel_tol = rel_tol, abs_tol = abs_tol): continue
                if len(col1) > 16:
                    arr1 = repr(col1[:16])[:-1] + ", ...]"
                    arr2 = repr(col2[:16])[:-1] + ", ...]"
                else:
                    arr1 = repr(col1)
                    arr2 = repr(col2)
>               raise AssertionError(
                    "The frames have different data in column %d `%s` at "
                    "index %d: LHS has %r, and RHS has %r\n"
                    "  LHS = %s\n"
                    "  RHS = %s\n"
                    % (i, frame1.names[i], j, val1, val2, arr1, arr2))
E               AssertionError: The frames have different data in column 0 `A` at index 0: LHS has 1, and RHS has None
E                 LHS = [1, 1, 1, 1, 2, 3, 3, 5, None]
E                 RHS = [None, 1, 1, 1, 1, 2, 3, 3, 5]
abs_tol    = 1e-07
arr1       = '[1, 1, 1, 1, 2, 3, 3, 5, None]'
arr2       = '[None, 1, 1, 1, 1, 2, 3, 3, 5]'
col1       = [1, 1, 1, 1, 2, 3, 3, 5, None]
col2       = [None, 1, 1, 1, 1, 2, 3, 3, 5]
data1      = [[1, 1, 1, 1, 2, 3, 3, 5, None], ['a', None, 'b', 'h', 'a', 'c', 'f', 'b', 'd']]
data2      = [[None, 1, 1, 1, 1, 2, 3, 3, 5], ['d', 'a', None, 'b', 'h', 'a', 'c', 'f', 'b']]
frame1     = <Frame#1afd0932e80 9x2>
frame2     = <Frame#1afd0932eb0 9x2>
i          = 0
j          = 0
rel_tol    = 1e-07
val1       = 1
val2       = None
tests\__init__.py:140: AssertionError
_________________________ test_fillna_grouped_column __________________________
    def test_fillna_grouped_column():
        DT = dt.Frame([2, 1, None, 1, 2])
        DT_fillna = DT[:, [fillna(f[0], reverse=False),
                          fillna(f[0], reverse=True)], by(f[0])]
        DT_ref = dt.Frame([
                     [None, 1, 1, 2, 2],
                     [None, 1, 1, 2, 2],
                     [None, 1, 1, 2, 2]
                 ])
>       assert_equals(DT_fillna, DT_ref)
DT         = <Frame#1afd1e46dc0 5x1>
DT_fillna  = <Frame#1afd1e46d90 5x3>
DT_ref     = <Frame#1afd1e46c10 5x3>
tests\dt\test-fillna.py:234: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
frame1 = <Frame#1afd1e46d90 5x3>, frame2 = <Frame#1afd1e46c10 5x3>
rel_tol = 1e-07, abs_tol = 1e-07
    def assert_equals(frame1, frame2, rel_tol = 1e-7, abs_tol = None):
        """
        Helper function to assert that 2 frames are equal to each other.
        """
        frame_integrity_check(frame1)
        frame_integrity_check(frame2)
    
        # The default value of `abs_tol` is set to `rel_tol`
        if abs_tol is None:
            abs_tol = rel_tol
    
        assert frame1.shape == frame2.shape, (
            "The left frame has shape %r, while the right has shape %r"
            % (frame1.shape, frame2.shape))
    
        assert frame1.names == frame2.names, (
            "The left frame has names %r, while the right has names %r"
            % (frame1.names, frame2.names))
        assert frame1.types == frame2.types, (
            "The left frame has types %r, while the right has types %r"
            % (frame1.types, frame2.types))
        data1 = frame1.to_list()
        data2 = frame2.to_list()
        assert len(data1) == len(data2)  # shape check should ensure this
        for i in range(len(data1)):
            col1 = data1[i]
            col2 = data2[i]
            assert len(col1) == len(col2)
            for j in range(len(col1)):
                val1 = col1[j]
                val2 = col2[j]
                if val1 == val2: continue
                if isinstance(val1, float) and isinstance(val2, float):
                    if math.isclose(val1, val2, rel_tol = rel_tol, abs_tol = abs_tol): continue
                if len(col1) > 16:
                    arr1 = repr(col1[:16])[:-1] + ", ...]"
                    arr2 = repr(col2[:16])[:-1] + ", ...]"
                else:
                    arr1 = repr(col1)
                    arr2 = repr(col2)
>               raise AssertionError(
                    "The frames have different data in column %d `%s` at "
                    "index %d: LHS has %r, and RHS has %r\n"
                    "  LHS = %s\n"
                    "  RHS = %s\n"
                    % (i, frame1.names[i], j, val1, val2, arr1, arr2))
E               AssertionError: The frames have different data in column 0 `C0` at index 0: LHS has 1, and RHS has None
E                 LHS = [1, 1, 2, 2, None]
E                 RHS = [None, 1, 1, 2, 2]
abs_tol    = 1e-07
arr1       = '[1, 1, 2, 2, None]'
arr2       = '[None, 1, 1, 2, 2]'
col1       = [1, 1, 2, 2, None]
col2       = [None, 1, 1, 2, 2]
data1      = [[1, 1, 2, 2, None], [1, 1, 2, 2, None], [1, 1, 2, 2, None]]
data2      = [[None, 1, 1, 2, 2], [None, 1, 1, 2, 2], [None, 1, 1, 2, 2]]
frame1     = <Frame#1afd1e46d90 5x3>
frame2     = <Frame#1afd1e46c10 5x3>
i          = 0
j          = 0
rel_tol    = 1e-07
val1       = 1
val2       = None
tests\__init__.py:140: AssertionError
```

--- a/ci/appveyor.yml
+++ b/ci/appveyor.yml
@@ -1,7 +1,7 @@
 image:
   - Visual Studio 2019
-  - macos-monterey
-  - Ubuntu
+  # - macos-monterey
+  # - Ubuntu
 
 init:
   # Uncomment the line below to enable RDP access on AppVeyor
@@ -182,123 +182,123 @@ build_script:
     echo "DT_BUILD_SUFFIX = $env:DT_BUILD_SUFFIX"
 
 
+    # # =======================================================================
+    # #   Build and test wheel for Python 3.7
+    # # =======================================================================
+
+    # $env:PATH = "C:/Python37-x64;C:/Python37-x64/Scripts;$DEFAULT_PATH"
+
+    # python -V
+
+    # python ci/ext.py wheel
+
+    # $DT_WHEEL = ls dist/*-cp37-*.whl
+
+    # echo "DT_WHEEL = $DT_WHEEL"
+
+    # echo "----- _build_info.py for Python 3.7 ------------------------------"
+
+    # cat src/datatable/_build_info.py
+
+    # echo "------------------------------------------------------------------"
+
+    # python -m pip install --upgrade pip
+
+    # python -m pip install $DT_WHEEL
+
+    # python -m pip install pytest docutils pandas pyarrow
+
+    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+
+    # if(!$?) { Exit $LASTEXITCODE }
+
+    # python -m pip uninstall -y $DT_WHEEL
+
+
+
+    # # =======================================================================
+    # #   Build and test wheel for Python 3.8
+    # # =======================================================================
+
+    # $env:PATH = "C:/Python38-x64;C:/Python38-x64/Scripts;$DEFAULT_PATH"
+
+    # python -V
+
+    # python ci/ext.py wheel
+
+    # $DT_WHEEL = ls dist/*-cp38-*.whl
+
+    # echo "DT_WHEEL = $DT_WHEEL"
+
+    # echo "----- _build_info.py for Python 3.8 ------------------------------"
+
+    # cat src/datatable/_build_info.py
+
+    # echo "------------------------------------------------------------------"
+
+    # python -m pip install --upgrade pip
+
+    # python -m pip install $DT_WHEEL
+
+    # python -m pip install pytest docutils pandas pyarrow
+
+    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+
+    # if(!$?) { Exit $LASTEXITCODE }
+
+    # python -m pip uninstall -y $DT_WHEEL
+
+
+
+    # # =======================================================================
+    # #   Build and test wheel for Python 3.9
+    # # =======================================================================
+
+    # $env:PATH = "C:/Python39-x64;C:/Python39-x64/Scripts;$DEFAULT_PATH"
+
+    # python -V
+
+    # python ci/ext.py wheel
+
+    # $DT_WHEEL = ls dist/*-cp39-*.whl
+
+    # echo "DT_WHEEL = $DT_WHEEL"
+
+    # echo "----- _build_info.py for Python 3.9 ------------------------------"
+
+    # cat src/datatable/_build_info.py
+
+    # echo "------------------------------------------------------------------"
+
+    # python -m pip install --upgrade pip
+
+    # python -m pip install $DT_WHEEL
+
+    # python -m pip install pytest docutils pandas pyarrow
+
+    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+
+    # if(!$?) { Exit $LASTEXITCODE }
+
+    # python -m pip uninstall -y $DT_WHEEL
+
+
+
     # =======================================================================
-    #   Build and test wheel for Python 3.7
-    # =======================================================================
-
-    $env:PATH = "C:/Python37-x64;C:/Python37-x64/Scripts;$DEFAULT_PATH"
-
-    python -V
-
-    python ci/ext.py wheel
-
-    $DT_WHEEL = ls dist/*-cp37-*.whl
-
-    echo "DT_WHEEL = $DT_WHEEL"
-
-    echo "----- _build_info.py for Python 3.7 ------------------------------"
-
-    cat src/datatable/_build_info.py
-
-    echo "------------------------------------------------------------------"
-
-    python -m pip install --upgrade pip
-
-    python -m pip install $DT_WHEEL
-
-    python -m pip install pytest docutils pandas pyarrow
-
-    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
-
-    if(!$?) { Exit $LASTEXITCODE }
-
-    python -m pip uninstall -y $DT_WHEEL
-
-
-
-    # =======================================================================
-    #   Build and test wheel for Python 3.8
+    #   Build and test debug wheel for Python 3.8
     # =======================================================================
 
     $env:PATH = "C:/Python38-x64;C:/Python38-x64/Scripts;$DEFAULT_PATH"
 
     python -V
 
-    python ci/ext.py wheel
-
-    $DT_WHEEL = ls dist/*-cp38-*.whl
-
-    echo "DT_WHEEL = $DT_WHEEL"
-
-    echo "----- _build_info.py for Python 3.8 ------------------------------"
-
-    cat src/datatable/_build_info.py
-
-    echo "------------------------------------------------------------------"
-
-    python -m pip install --upgrade pip
-
-    python -m pip install $DT_WHEEL
-
-    python -m pip install pytest docutils pandas pyarrow
-
-    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
-
-    if(!$?) { Exit $LASTEXITCODE }
-
-    python -m pip uninstall -y $DT_WHEEL
-
-
-
-    # =======================================================================
-    #   Build and test wheel for Python 3.9
-    # =======================================================================
-
-    $env:PATH = "C:/Python39-x64;C:/Python39-x64/Scripts;$DEFAULT_PATH"
-
-    python -V
-
-    python ci/ext.py wheel
-
-    $DT_WHEEL = ls dist/*-cp39-*.whl
-
-    echo "DT_WHEEL = $DT_WHEEL"
-
-    echo "----- _build_info.py for Python 3.9 ------------------------------"
-
-    cat src/datatable/_build_info.py
-
-    echo "------------------------------------------------------------------"
-
-    python -m pip install --upgrade pip
-
-    python -m pip install $DT_WHEEL
-
-    python -m pip install pytest docutils pandas pyarrow
-
-    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
-
-    if(!$?) { Exit $LASTEXITCODE }
-
-    python -m pip uninstall -y $DT_WHEEL
-
-
-
-    # =======================================================================
-    #   Build and test debug wheel for Python 3.9
-    # =======================================================================
-
-    $env:PATH = "C:/Python39-x64;C:/Python39-x64/Scripts;$DEFAULT_PATH"
-
-    python -V
-
     python ci/ext.py debugwheel
 
-    $DT_WHEEL = ls dist/*debug-cp39-*.whl
+    $DT_WHEEL = ls dist/*debug-cp38-*.whl
 
     echo "DT_WHEEL = $DT_WHEEL"
 
-    echo "----- _build_info.py for Python 3.9 debug wheel ------------------"
+    echo "----- _build_info.py for Python 3.8 debug wheel ------------------"
 
     cat src/datatable/_build_info.py
 
@@ -318,35 +318,35 @@ build_script:
 
 
 
-    # =======================================================================
-    #   Build and test wheel for Python 3.10
-    # =======================================================================
+    # # =======================================================================
+    # #   Build and test wheel for Python 3.10
+    # # =======================================================================
 
-    $env:PATH = "C:/Python310-x64;C:/Python310-x64/Scripts;$DEFAULT_PATH"
+    # $env:PATH = "C:/Python310-x64;C:/Python310-x64/Scripts;$DEFAULT_PATH"
 
-    python -V
+    # python -V
 
-    python ci/ext.py wheel
+    # python ci/ext.py wheel
 
-    $DT_WHEEL = ls dist/*-cp310-*.whl
+    # $DT_WHEEL = ls dist/*-cp310-*.whl
 
-    echo "DT_WHEEL = $DT_WHEEL"
+    # echo "DT_WHEEL = $DT_WHEEL"
 
-    echo "----- _build_info.py for Python 3.10 ------------------------------"
+    # echo "----- _build_info.py for Python 3.10 ------------------------------"
 
-    cat src/datatable/_build_info.py
+    # cat src/datatable/_build_info.py
 
-    echo "------------------------------------------------------------------"
+    # echo "------------------------------------------------------------------"
 
-    python -m pip install --upgrade pip
+    # python -m pip install --upgrade pip
 
-    python -m pip install $DT_WHEEL
+    # python -m pip install $DT_WHEEL
 
-    python -m pip install pytest docutils pandas pyarrow
+    # python -m pip install pytest docutils pandas pyarrow
 
-    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
 
-    if(!$?) { Exit $LASTEXITCODE }
+    # if(!$?) { Exit $LASTEXITCODE }
 
-    python -m pip uninstall -y $DT_WHEEL
+    # python -m pip uninstall -y $DT_WHEEL
 

--- a/ci/appveyor.yml
+++ b/ci/appveyor.yml
@@ -1,7 +1,7 @@
 image:
   - Visual Studio 2019
-  # - macos-monterey
-  # - Ubuntu
+  - macos-monterey
+  - Ubuntu
 
 init:
   # Uncomment the line below to enable RDP access on AppVeyor

--- a/ci/appveyor.yml
+++ b/ci/appveyor.yml
@@ -182,105 +182,72 @@ build_script:
     echo "DT_BUILD_SUFFIX = $env:DT_BUILD_SUFFIX"
 
 
-    # # =======================================================================
-    # #   Build and test wheel for Python 3.7
-    # # =======================================================================
 
-    # $env:PATH = "C:/Python37-x64;C:/Python37-x64/Scripts;$DEFAULT_PATH"
+    # =======================================================================
+    #   Build and test wheel for Python 3.7
+    # =======================================================================
 
-    # python -V
+    $env:PATH = "C:/Python37-x64;C:/Python37-x64/Scripts;$DEFAULT_PATH"
 
-    # python ci/ext.py wheel
+    python -V
 
-    # $DT_WHEEL = ls dist/*-cp37-*.whl
+    python ci/ext.py wheel
 
-    # echo "DT_WHEEL = $DT_WHEEL"
+    $DT_WHEEL = ls dist/*-cp37-*.whl
 
-    # echo "----- _build_info.py for Python 3.7 ------------------------------"
+    echo "DT_WHEEL = $DT_WHEEL"
 
-    # cat src/datatable/_build_info.py
+    echo "----- _build_info.py for Python 3.7 ------------------------------"
 
-    # echo "------------------------------------------------------------------"
+    cat src/datatable/_build_info.py
 
-    # python -m pip install --upgrade pip
+    echo "------------------------------------------------------------------"
 
-    # python -m pip install $DT_WHEEL
+    python -m pip install --upgrade pip
 
-    # python -m pip install pytest docutils pandas pyarrow
+    python -m pip install $DT_WHEEL
 
-    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+    python -m pip install pytest docutils pandas pyarrow
 
-    # if(!$?) { Exit $LASTEXITCODE }
+    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
 
-    # python -m pip uninstall -y $DT_WHEEL
+    if(!$?) { Exit $LASTEXITCODE }
 
-
-
-    # # =======================================================================
-    # #   Build and test wheel for Python 3.8
-    # # =======================================================================
-
-    # $env:PATH = "C:/Python38-x64;C:/Python38-x64/Scripts;$DEFAULT_PATH"
-
-    # python -V
-
-    # python ci/ext.py wheel
-
-    # $DT_WHEEL = ls dist/*-cp38-*.whl
-
-    # echo "DT_WHEEL = $DT_WHEEL"
-
-    # echo "----- _build_info.py for Python 3.8 ------------------------------"
-
-    # cat src/datatable/_build_info.py
-
-    # echo "------------------------------------------------------------------"
-
-    # python -m pip install --upgrade pip
-
-    # python -m pip install $DT_WHEEL
-
-    # python -m pip install pytest docutils pandas pyarrow
-
-    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
-
-    # if(!$?) { Exit $LASTEXITCODE }
-
-    # python -m pip uninstall -y $DT_WHEEL
+    python -m pip uninstall -y $DT_WHEEL
 
 
 
-    # # =======================================================================
-    # #   Build and test wheel for Python 3.9
-    # # =======================================================================
+    # =======================================================================
+    #   Build and test wheel for Python 3.8
+    # =======================================================================
 
-    # $env:PATH = "C:/Python39-x64;C:/Python39-x64/Scripts;$DEFAULT_PATH"
+    $env:PATH = "C:/Python38-x64;C:/Python38-x64/Scripts;$DEFAULT_PATH"
 
-    # python -V
+    python -V
 
-    # python ci/ext.py wheel
+    python ci/ext.py wheel
 
-    # $DT_WHEEL = ls dist/*-cp39-*.whl
+    $DT_WHEEL = ls dist/*-cp38-*.whl
 
-    # echo "DT_WHEEL = $DT_WHEEL"
+    echo "DT_WHEEL = $DT_WHEEL"
 
-    # echo "----- _build_info.py for Python 3.9 ------------------------------"
+    echo "----- _build_info.py for Python 3.8 ------------------------------"
 
-    # cat src/datatable/_build_info.py
+    cat src/datatable/_build_info.py
 
-    # echo "------------------------------------------------------------------"
+    echo "------------------------------------------------------------------"
 
-    # python -m pip install --upgrade pip
+    python -m pip install --upgrade pip
 
-    # python -m pip install $DT_WHEEL
+    python -m pip install $DT_WHEEL
 
-    # python -m pip install pytest docutils pandas pyarrow
+    python -m pip install pytest docutils pandas pyarrow
 
-    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
 
-    # if(!$?) { Exit $LASTEXITCODE }
+    if(!$?) { Exit $LASTEXITCODE }
 
-    # python -m pip uninstall -y $DT_WHEEL
+    python -m pip uninstall -y $DT_WHEEL
 
 
 
@@ -318,35 +285,69 @@ build_script:
 
 
 
-    # # =======================================================================
-    # #   Build and test wheel for Python 3.10
-    # # =======================================================================
+    # =======================================================================
+    #   Build and test wheel for Python 3.9
+    # =======================================================================
 
-    # $env:PATH = "C:/Python310-x64;C:/Python310-x64/Scripts;$DEFAULT_PATH"
+    $env:PATH = "C:/Python39-x64;C:/Python39-x64/Scripts;$DEFAULT_PATH"
 
-    # python -V
+    python -V
 
-    # python ci/ext.py wheel
+    python ci/ext.py wheel
 
-    # $DT_WHEEL = ls dist/*-cp310-*.whl
+    $DT_WHEEL = ls dist/*-cp39-*.whl
 
-    # echo "DT_WHEEL = $DT_WHEEL"
+    echo "DT_WHEEL = $DT_WHEEL"
 
-    # echo "----- _build_info.py for Python 3.10 ------------------------------"
+    echo "----- _build_info.py for Python 3.9 ------------------------------"
 
-    # cat src/datatable/_build_info.py
+    cat src/datatable/_build_info.py
 
-    # echo "------------------------------------------------------------------"
+    echo "------------------------------------------------------------------"
 
-    # python -m pip install --upgrade pip
+    python -m pip install --upgrade pip
 
-    # python -m pip install $DT_WHEEL
+    python -m pip install $DT_WHEEL
 
-    # python -m pip install pytest docutils pandas pyarrow
+    python -m pip install pytest docutils pandas pyarrow
 
-    # python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
 
-    # if(!$?) { Exit $LASTEXITCODE }
+    if(!$?) { Exit $LASTEXITCODE }
 
-    # python -m pip uninstall -y $DT_WHEEL
+    python -m pip uninstall -y $DT_WHEEL
+
+
+
+    # =======================================================================
+    #   Build and test wheel for Python 3.10
+    # =======================================================================
+
+    $env:PATH = "C:/Python310-x64;C:/Python310-x64/Scripts;$DEFAULT_PATH"
+
+    python -V
+
+    python ci/ext.py wheel
+
+    $DT_WHEEL = ls dist/*-cp310-*.whl
+
+    echo "DT_WHEEL = $DT_WHEEL"
+
+    echo "----- _build_info.py for Python 3.10 ------------------------------"
+
+    cat src/datatable/_build_info.py
+
+    echo "------------------------------------------------------------------"
+
+    python -m pip install --upgrade pip
+
+    python -m pip install $DT_WHEEL
+
+    python -m pip install pytest docutils pandas pyarrow
+
+    python -m pytest -ra --maxfail=10 -Werror -vv --showlocals ./tests/
+
+    if(!$?) { Exit $LASTEXITCODE }
+
+    python -m pip uninstall -y $DT_WHEEL
 

--- a/docs/releases/v1.1.0.rst
+++ b/docs/releases/v1.1.0.rst
@@ -53,6 +53,8 @@
     -[fix] Fixed :meth:`.to_csv()` to quote missing values when
         `quoting="all"` is specified. [#3340]
 
+    -[fix] Fixed groupby behaviour on columns that contain missing values. [#3331]
+
     -[api] Converting a column of :attr:`void <dt.Type.void>` type into pandas
         now produces a pandas ``object`` column filled with ``None``s. Converting
         such column back into datatable produces a ``void`` column again. [#3063]

--- a/src/core/expr/eval_context.cc
+++ b/src/core/expr/eval_context.cc
@@ -45,6 +45,7 @@ EvalContext::EvalContext(DataTable* dt, EvalMode evalmode)
   frames_.emplace_back(dt, RowIndex(), /* natural_join= */ false);
   eval_mode_ = evalmode;
   add_groupby_columns_ = true;
+  na_position_ = NaPosition::FIRST;
 }
 
 

--- a/src/core/expr/eval_context.cc
+++ b/src/core/expr/eval_context.cc
@@ -45,7 +45,7 @@ EvalContext::EvalContext(DataTable* dt, EvalMode evalmode)
   frames_.emplace_back(dt, RowIndex(), /* natural_join= */ false);
   eval_mode_ = evalmode;
   add_groupby_columns_ = true;
-  na_position_ = NaPosition::FIRST;
+  na_position_ = NaPosition::FIRST; // default for groupby
 }
 
 

